kibana-reports/public/components/main/main_utils.tsx

```diff
@@ -159,16 +160,23 @@ export const generateReport = async (metadata, httpClient) => {
     })
     .catch((error) => {
       console.log('error on generating report:', error);
+      if (error.body.statusCode === 403) {
+        permissionsError = true;
+      }
       status = false;
     });
-  return status;
+  return {
+    status: status,
+    permissionsError: permissionsError,
```

Comment:
Is it possible to leave only one status and make it generic? I assume it will need some refactor for toast, maybe not something we have time to do for this PR

Reply:
We could change status from a boolean to a string, that's an option we could use as well. But we only use the return value of this function in one place anyway

Reply:
I mean for the creation of error toast, correct me if I am wrong. Currently you defined two sets of error toast, one for general error, and one for no permission error. By "generic" I mean to merge those two into one, maybe createErrorToast(), and control the text shown by passing corresponding parameter(a string)

Reply:
I see what you mean now, will change